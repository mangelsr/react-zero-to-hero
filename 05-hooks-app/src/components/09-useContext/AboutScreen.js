import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser({});
    };

    return (
        <div>
            <h1>About Screen</h1>
            <hr/>

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>

            <button
                onClick={handleClick}
                className="btn btn-outline-danger">
                Logout
            </button>
        </div>
    )
}
