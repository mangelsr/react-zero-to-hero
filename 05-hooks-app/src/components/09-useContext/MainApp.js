import React, { useEffect, useState } from 'react';
import { AppRouter } from './AppRouter';

import { UserContext } from './UserContext';
import './style.css';


let initialUser = JSON.parse(localStorage.getItem('user'));
if (initialUser === null) {
    initialUser = {};
}

export const MainApp = () => {

    const [user, setUser] = useState(initialUser);

    useEffect(() => {
        localStorage.setItem('user', JSON.stringify(user));
    }, [user]);

    return (
        <UserContext.Provider value={{ user, setUser }}>
            <AppRouter/>
        </UserContext.Provider>   
    )
}
