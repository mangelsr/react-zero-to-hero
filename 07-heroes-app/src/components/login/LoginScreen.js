import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';


export const LoginScreen = ({ history }) => {

    const { dispatch } = useContext(AuthContext);
    
    const handleLogin = () => {
        dispatch({
            type: types.login,
            payload: {
                name: 'Miguel'
            }
        });
        const lastPath = localStorage.getItem('lastpath') || '/';
        history.replace(lastPath);
    };

    return (
        <div className="container mt-5">
            <h1>Login Screen</h1>
            <hr/>
            <button
                onClick={handleLogin}
                className="btn btn-outline-success"
                >
                Login
            </button>
        </div>
    )
}
