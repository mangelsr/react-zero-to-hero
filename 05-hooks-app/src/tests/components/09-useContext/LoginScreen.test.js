import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';

import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';


describe('LoginScreen component tests', () => {

    const setUser = jest.fn();
    
    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should execute setUser with expected argument', () => {
        wrapper.find('button').simulate('click');
        expect(setUser).toBeCalledWith({
            id:123,
            name:'Miguel'
        });
    });
    
});