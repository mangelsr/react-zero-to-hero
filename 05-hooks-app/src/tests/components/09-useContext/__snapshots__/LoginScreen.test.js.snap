// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen component tests should render the component 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
